zcash_client_sqlite/src/wallet/init/migrations/nullifier_map.rs

```diff
@@ -0,0 +1,72 @@
+//! This migration adds a table for storing mappings from nullifiers to the transaction
+//! they are revealed in.
+
+use std::collections::HashSet;
+
+use schemer_rusqlite::RusqliteMigration;
+use tracing::debug;
+use uuid::Uuid;
+
+use crate::wallet::init::WalletMigrationError;
+
+use super::received_notes_nullable_nf;
+
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_u128(0xe2d71ac5_6a44_4c6b_a9a0_6d0a79d355f1);
+
+pub(super) struct Migration;
+
+impl schemer::Migration for Migration {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        [received_notes_nullable_nf::MIGRATION_ID]
+            .into_iter()
+            .collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Adds a lookup table for nullifiers we've observed on-chain that we haven't confirmed are not ours."
+    }
+}
+
+impl RusqliteMigration for Migration {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), Self::Error> {
+        // We don't enforce any foreign key constraint to the blocks table, to allow
+        // loading the nullifier map separately from block scanning.
+        debug!("Creating tables for nullifier map");
+        transaction.execute_batch(
+            "CREATE TABLE tx_locator_map (
```

Comment:
nit: I don't understand why these tables have the `_map` suffix; we don't use that convention elsewhere.